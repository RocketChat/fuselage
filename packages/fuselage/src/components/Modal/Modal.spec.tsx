import React from 'react';
import ReactDOM from 'react-dom';

import { Modal } from '../..';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Modal />, div);
  ReactDOM.unmountComponentAtNode(div);
});
