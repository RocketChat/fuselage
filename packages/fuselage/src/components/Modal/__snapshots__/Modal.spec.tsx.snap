// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Modal Component] renders _WithAnnotation without crashing 1`] = `
<body>
  <div>
    <dialog
      aria-modal="true"
      class="rcx-box rcx-box--full rcx-modal"
    >
      <div
        class="rcx-box rcx-box--full  rcx-modal__inner rcx-css-1ppp6za"
      >
        <header
          class="rcx-box rcx-box--full rcx-modal__header"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__header-inner"
          >
            <div
              class="rcx-box rcx-box--full rcx-modal__header-text  rcx-css-trljwa rcx-css-lma364"
            >
              <h2
                class="rcx-box rcx-box--full rcx-modal__title"
              >
                Modal Header
              </h2>
            </div>
            <button
              aria-label="cross"
              class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button  rcx-css-trljwa rcx-css-lma364"
              type="button"
            >
              <i
                aria-hidden="true"
                class="rcx-box rcx-box--full rcx-icon--name-cross rcx-icon rcx-css-4pvxx3"
              >
                
              </i>
            </button>
          </div>
        </header>
        <div
          class="rcx-box rcx-box--full rcx-modal__content  rcx-css-1vw7itl"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__content-wrapper rcx-css-56vo9k"
          >
            Modal Body
          </div>
        </div>
        <div
          class="rcx-box rcx-box--full rcx-modal__footer rcx-css-rzpqmp"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__footer-annotation"
          >
            Anototation
          </div>
          <div
            class="rcx-box rcx-box--full rcx-button-group--align-end rcx-button-group rcx-css-94c313"
            role="group"
          >
            <button
              class="rcx-box rcx-box--full rcx-button  rcx-button-group__item"
              type="button"
            >
              <span
                class="rcx-button--content"
              >
                Cancel
              </span>
            </button>
            <button
              class="rcx-box rcx-box--full rcx-button--primary rcx-button  rcx-button-group__item"
              type="button"
            >
              <span
                class="rcx-button--content"
              >
                Submit
              </span>
            </button>
          </div>
        </div>
      </div>
    </dialog>
  </div>
</body>
`;

exports[`[Modal Component] renders _WithForm without crashing 1`] = `
<body>
  <div>
    <dialog
      aria-modal="true"
      class="rcx-box rcx-box--full rcx-modal"
    >
      <form
        class="rcx-box rcx-box--full  rcx-modal__inner rcx-css-1ppp6za"
      >
        <header
          class="rcx-box rcx-box--full rcx-modal__header"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__header-inner"
          >
            <div
              class="rcx-box rcx-box--full rcx-modal__header-text  rcx-css-trljwa rcx-css-lma364"
            >
              <h2
                class="rcx-box rcx-box--full rcx-modal__title"
              >
                Modal Header
              </h2>
            </div>
            <button
              aria-label="cross"
              class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button  rcx-css-trljwa rcx-css-lma364"
              type="button"
            >
              <i
                aria-hidden="true"
                class="rcx-box rcx-box--full rcx-icon--name-cross rcx-icon rcx-css-4pvxx3"
              >
                
              </i>
            </button>
          </div>
        </header>
        <div
          class="rcx-box rcx-box--full rcx-modal__content  rcx-css-1vw7itl"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__content-wrapper rcx-css-56vo9k"
          >
            <fieldset
              class="rcx-box rcx-box--full rcx-field-group"
              role="group"
            >
              <div
                class="rcx-box rcx-box--full rcx-field  rcx-field-group__item"
              >
                <label
                  class="rcx-box rcx-box--full rcx-label  rcx-box rcx-box--full rcx-field__label"
                >
                  Label
                </label>
                <span
                  class="rcx-box rcx-box--full rcx-field__row"
                >
                  <input
                    class="rcx-box rcx-box--full rcx-box--animated rcx-input-box--type-text rcx-input-box"
                    placeholder="Placeholder"
                    size="1"
                    type="text"
                  />
                </span>
              </div>
            </fieldset>
          </div>
        </div>
        <div
          class="rcx-box rcx-box--full rcx-modal__footer rcx-css-17mu816"
        >
          <div
            class="rcx-box rcx-box--full rcx-button-group--align-end rcx-button-group rcx-css-94c313"
            role="group"
          >
            <button
              class="rcx-box rcx-box--full rcx-button  rcx-button-group__item"
              type="button"
            >
              <span
                class="rcx-button--content"
              >
                Cancel
              </span>
            </button>
            <button
              class="rcx-box rcx-box--full rcx-button--primary rcx-button  rcx-button-group__item"
              type="submit"
            >
              <span
                class="rcx-button--content"
              >
                Submit
              </span>
            </button>
          </div>
        </div>
      </form>
    </dialog>
  </div>
</body>
`;

exports[`[Modal Component] renders _WithHeroImage without crashing 1`] = `
<body>
  <div>
    <dialog
      aria-modal="true"
      class="rcx-box rcx-box--full rcx-modal"
    >
      <div
        class="rcx-box rcx-box--full  rcx-modal__inner rcx-css-1ppp6za"
      >
        <header
          class="rcx-box rcx-box--full rcx-modal__header"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__header-inner"
          >
            <div
              class="rcx-box rcx-box--full rcx-modal__header-text  rcx-css-trljwa rcx-css-lma364"
            >
              <h2
                class="rcx-box rcx-box--full rcx-modal__title"
              >
                Modal Header
              </h2>
            </div>
            <button
              aria-label="cross"
              class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button  rcx-css-trljwa rcx-css-lma364"
              type="button"
            >
              <i
                aria-hidden="true"
                class="rcx-box rcx-box--full rcx-icon--name-cross rcx-icon rcx-css-4pvxx3"
              >
                
              </i>
            </button>
          </div>
        </header>
        <div
          class="rcx-box rcx-box--full rcx-modal__content  rcx-css-1vw7itl"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__content-wrapper rcx-css-56vo9k"
          >
            <figure
              class="rcx-modal__hero-image-wrapper"
            >
              <img
                class="rcx-box rcx-box--full rcx-modal__hero-image"
                src="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wBDAAgGBgcGBQgHBwcJCQgKDBQNDAsLDBkSEw8UHRofHh0aHBwgJC4nICIsIxwcKDcpLDAxNDQ0Hyc5PTgyPC4zNDL/2wBDAQkJCQwLDBgNDRgyIRwhMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjL/wAARCAAoACgDASIAAhEBAxEB/8QAGwAAAgIDAQAAAAAAAAAAAAAAAAcEBgIDBQj/xAAuEAACAQQAAwcEAQUAAAAAAAABAgMABAUREiExBhMUIkFRYQcWcYGhFTJSgpH/xAAYAQADAQEAAAAAAAAAAAAAAAACAwQBAP/EAB4RAAIBBQEBAQAAAAAAAAAAAAABAgMREiExE0HR/9oADAMBAAIRAxEAPwBuXuIkhBuMe5ib/AHQP49q4L3mLitryTLTSpOiHQI5k/HzXa/qbFOEudVTu1dumWvcTaNCZYZ7vU6g6LxqjOU/24dfs1Ouh9FnkMpd3Reeyx83hAxZZEhkdV9/MBrX71WGPvJcqrJBGveKATtuXXqNU0pu02bTHXD/AGvJAluyxxRd6F4x00o+NdKoVrjbzJdvVe1t5cVLc2ck8qjnohgpPtz2v7G6JtPQ2VJwjlcw+37mchpnK6GtIuv5NFWeTsLNPvxWTvpfjvOEfwKKzEVkSct2vscS/BIzSN0YRkeX81UpPqO8masJETu7OOccY4dswYFQeftv096XV5knuJGdm2T1+agvMXj8jEaHX905QihabvcbuS7X566mLWLwSY8PuRnk/u4eZ0deTl71Ef6hY+0yM88TzeNZY4luYwpVYyduOfrvhPTnr0pXSX9y5mCsyJMdyxxvwq599em+taItqCSNc90ChvZRUruUcT0JiO18Elpk7t8v41LWzacxkBSuvjQ/FFJayjDWrCTepAQ2vUH0oo/Jk3ovpwJJeVCP5CN+lFFaaMqy+nAyuChvrTI2kN9JAsi2ZOy4IBHMnkSCP+iqBexSWdxLazoUljJVlPUH2oorkV10pRc7b1zXb/hZOzuJvM86QWEXeELxOzHSIPcmiiiunVlF2RNTpRkrs//Z"
              />
            </figure>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse non urna posuere, tempor urna nec, lacinia lacus. Vivamus ac lobortis arcu. Morbi malesuada, diam sed congue aliquet, dui elit tincidunt leo, a fermentum ante augue nec ex. Vestibulum feugiat aliquam sem vel porta. Praesent varius aliquet consequat. Mauris id nisl egestas, egestas ipsum sit amet, vestibulum elit. Cras vel dapibus lacus. Sed congue interdum lobortis. In vitae consectetur enim, eu varius leo. Quisque rhoncus nulla a rhoncus lobortis. Sed eu nulla libero. Donec lacus ante, vehicula eget eros molestie, ullamcorper tincidunt arcu. Suspendisse eget pulvinar lacus.
          </div>
        </div>
        <div
          class="rcx-box rcx-box--full rcx-modal__footer rcx-css-17mu816"
        >
          <div
            class="rcx-box rcx-box--full rcx-button-group--align-end rcx-button-group rcx-css-94c313"
            role="group"
          >
            <button
              class="rcx-box rcx-box--full rcx-button  rcx-button-group__item"
              type="button"
            >
              <span
                class="rcx-button--content"
              >
                Cancel
              </span>
            </button>
            <button
              class="rcx-box rcx-box--full rcx-button--primary rcx-button  rcx-button-group__item"
              type="button"
            >
              <span
                class="rcx-button--content"
              >
                Submit
              </span>
            </button>
          </div>
        </div>
      </div>
    </dialog>
  </div>
</body>
`;

exports[`[Modal Component] renders _WithIcon without crashing 1`] = `
<body>
  <div>
    <dialog
      aria-modal="true"
      class="rcx-box rcx-box--full rcx-modal"
    >
      <div
        class="rcx-box rcx-box--full  rcx-modal__inner rcx-css-1ppp6za"
      >
        <header
          class="rcx-box rcx-box--full rcx-modal__header"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__header-inner"
          >
            <div
              class="rcx-box rcx-box--full  rcx-css-trljwa rcx-css-1xb12qx"
            >
              <i
                aria-hidden="true"
                class="rcx-box rcx-box--full rcx-icon--name-info rcx-icon rcx-css-zozvgz"
              >
                
              </i>
            </div>
            <div
              class="rcx-box rcx-box--full rcx-modal__header-text  rcx-css-trljwa rcx-css-lma364"
            >
              <h2
                class="rcx-box rcx-box--full rcx-modal__title"
              >
                Modal Header
              </h2>
            </div>
            <button
              aria-label="cross"
              class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button  rcx-css-trljwa rcx-css-lma364"
              type="button"
            >
              <i
                aria-hidden="true"
                class="rcx-box rcx-box--full rcx-icon--name-cross rcx-icon rcx-css-4pvxx3"
              >
                
              </i>
            </button>
          </div>
        </header>
        <div
          class="rcx-box rcx-box--full rcx-modal__content  rcx-css-1vw7itl"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__content-wrapper rcx-css-56vo9k"
          >
            Modal Body
          </div>
        </div>
        <div
          class="rcx-box rcx-box--full rcx-modal__footer rcx-css-17mu816"
        >
          <div
            class="rcx-box rcx-box--full rcx-button-group--align-end rcx-button-group rcx-css-94c313"
            role="group"
          >
            <button
              class="rcx-box rcx-box--full rcx-button  rcx-button-group__item"
              type="button"
            >
              <span
                class="rcx-button--content"
              >
                Cancel
              </span>
            </button>
            <button
              class="rcx-box rcx-box--full rcx-button--primary rcx-button  rcx-button-group__item"
              type="button"
            >
              <span
                class="rcx-button--content"
              >
                Submit
              </span>
            </button>
          </div>
        </div>
      </div>
    </dialog>
  </div>
</body>
`;

exports[`[Modal Component] renders _WithIconAndTagline without crashing 1`] = `
<body>
  <div>
    <dialog
      aria-modal="true"
      class="rcx-box rcx-box--full rcx-modal"
    >
      <div
        class="rcx-box rcx-box--full  rcx-modal__inner rcx-css-1ppp6za"
      >
        <header
          class="rcx-box rcx-box--full rcx-modal__header"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__header-inner"
          >
            <div
              class="rcx-box rcx-box--full  rcx-css-trljwa rcx-css-15wh51c"
            >
              <i
                aria-hidden="true"
                class="rcx-box rcx-box--full rcx-icon--name-info rcx-icon rcx-css-zozvgz"
              >
                
              </i>
            </div>
            <div
              class="rcx-box rcx-box--full rcx-modal__header-text  rcx-css-trljwa rcx-css-lma364"
            >
              <div
                class="rcx-box rcx-box--full rcx-modal__tagline"
              >
                Tagline
              </div>
              <h2
                class="rcx-box rcx-box--full rcx-modal__title"
              >
                Modal Header
              </h2>
            </div>
            <button
              aria-label="cross"
              class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button  rcx-css-trljwa rcx-css-lma364"
              type="button"
            >
              <i
                aria-hidden="true"
                class="rcx-box rcx-box--full rcx-icon--name-cross rcx-icon rcx-css-4pvxx3"
              >
                
              </i>
            </button>
          </div>
        </header>
        <div
          class="rcx-box rcx-box--full rcx-modal__content  rcx-css-1vw7itl"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__content-wrapper rcx-css-56vo9k"
          >
            Modal Body
          </div>
        </div>
        <div
          class="rcx-box rcx-box--full rcx-modal__footer rcx-css-17mu816"
        >
          <div
            class="rcx-box rcx-box--full rcx-button-group--align-end rcx-button-group rcx-css-94c313"
            role="group"
          >
            <button
              class="rcx-box rcx-box--full rcx-button  rcx-button-group__item"
              type="button"
            >
              <span
                class="rcx-button--content"
              >
                Cancel
              </span>
            </button>
            <button
              class="rcx-box rcx-box--full rcx-button--primary rcx-button  rcx-button-group__item"
              type="button"
            >
              <span
                class="rcx-button--content"
              >
                Submit
              </span>
            </button>
          </div>
        </div>
      </div>
    </dialog>
  </div>
</body>
`;

exports[`[Modal Component] renders _WithTagline without crashing 1`] = `
<body>
  <div>
    <dialog
      aria-modal="true"
      class="rcx-box rcx-box--full rcx-modal"
    >
      <div
        class="rcx-box rcx-box--full  rcx-modal__inner rcx-css-1ppp6za"
      >
        <header
          class="rcx-box rcx-box--full rcx-modal__header"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__header-inner"
          >
            <div
              class="rcx-box rcx-box--full rcx-modal__header-text  rcx-css-trljwa rcx-css-lma364"
            >
              <div
                class="rcx-box rcx-box--full rcx-modal__tagline"
              >
                Tagline
              </div>
              <h2
                class="rcx-box rcx-box--full rcx-modal__title"
              >
                Modal Header
              </h2>
            </div>
            <button
              aria-label="cross"
              class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button  rcx-css-trljwa rcx-css-lma364"
              type="button"
            >
              <i
                aria-hidden="true"
                class="rcx-box rcx-box--full rcx-icon--name-cross rcx-icon rcx-css-4pvxx3"
              >
                
              </i>
            </button>
          </div>
        </header>
        <div
          class="rcx-box rcx-box--full rcx-modal__content  rcx-css-1vw7itl"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__content-wrapper rcx-css-56vo9k"
          >
            Modal Body
          </div>
        </div>
        <div
          class="rcx-box rcx-box--full rcx-modal__footer rcx-css-17mu816"
        >
          <div
            class="rcx-box rcx-box--full rcx-button-group--align-end rcx-button-group rcx-css-94c313"
            role="group"
          >
            <button
              class="rcx-box rcx-box--full rcx-button  rcx-button-group__item"
              type="button"
            >
              <span
                class="rcx-button--content"
              >
                Cancel
              </span>
            </button>
            <button
              class="rcx-box rcx-box--full rcx-button--primary rcx-button  rcx-button-group__item"
              type="button"
            >
              <span
                class="rcx-button--content"
              >
                Submit
              </span>
            </button>
          </div>
        </div>
      </div>
    </dialog>
  </div>
</body>
`;

exports[`[Modal Component] renders _WithThumb without crashing 1`] = `
<body>
  <div>
    <dialog
      aria-modal="true"
      class="rcx-box rcx-box--full rcx-modal"
    >
      <div
        class="rcx-box rcx-box--full  rcx-modal__inner rcx-css-1ppp6za"
      >
        <header
          class="rcx-box rcx-box--full rcx-modal__header"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__header-inner"
          >
            <div
              class="rcx-box rcx-box--full rcx-css-lma364"
            >
              <figure
                class="rcx-box rcx-box--full rcx-avatar rcx-avatar--x28 rcx-css-trljwa"
              >
                <img
                  class="rcx-avatar__element rcx-avatar__element--x28"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAMLCwgAAACH5BAAAAAAALAAAAAABAAEAAAICRAEAOw=="
                />
              </figure>
            </div>
            <div
              class="rcx-box rcx-box--full rcx-modal__header-text  rcx-css-trljwa rcx-css-lma364"
            >
              <h2
                class="rcx-box rcx-box--full rcx-modal__title"
              >
                Modal Header
              </h2>
            </div>
            <button
              aria-label="cross"
              class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button  rcx-css-trljwa rcx-css-lma364"
              type="button"
            >
              <i
                aria-hidden="true"
                class="rcx-box rcx-box--full rcx-icon--name-cross rcx-icon rcx-css-4pvxx3"
              >
                
              </i>
            </button>
          </div>
        </header>
        <div
          class="rcx-box rcx-box--full rcx-modal__content  rcx-css-1vw7itl"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__content-wrapper rcx-css-56vo9k"
          >
            Modal Body
          </div>
        </div>
        <div
          class="rcx-box rcx-box--full rcx-modal__footer rcx-css-17mu816"
        >
          <div
            class="rcx-box rcx-box--full rcx-button-group--align-end rcx-button-group rcx-css-94c313"
            role="group"
          >
            <button
              class="rcx-box rcx-box--full rcx-button  rcx-button-group__item"
              type="button"
            >
              <span
                class="rcx-button--content"
              >
                Cancel
              </span>
            </button>
            <button
              class="rcx-box rcx-box--full rcx-button--primary rcx-button  rcx-button-group__item"
              type="button"
            >
              <span
                class="rcx-button--content"
              >
                Submit
              </span>
            </button>
          </div>
        </div>
      </div>
    </dialog>
  </div>
</body>
`;

exports[`[Modal Component] renders Default without crashing 1`] = `
<body>
  <div>
    <dialog
      aria-modal="true"
      class="rcx-box rcx-box--full rcx-modal"
    >
      <div
        class="rcx-box rcx-box--full  rcx-modal__inner rcx-css-1ppp6za"
      >
        <header
          class="rcx-box rcx-box--full rcx-modal__header"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__header-inner"
          >
            <div
              class="rcx-box rcx-box--full rcx-modal__header-text  rcx-css-trljwa rcx-css-lma364"
            >
              <h2
                class="rcx-box rcx-box--full rcx-modal__title"
              >
                Modal Header
              </h2>
            </div>
            <button
              aria-label="cross"
              class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button  rcx-css-trljwa rcx-css-lma364"
              type="button"
            >
              <i
                aria-hidden="true"
                class="rcx-box rcx-box--full rcx-icon--name-cross rcx-icon rcx-css-4pvxx3"
              >
                
              </i>
            </button>
          </div>
        </header>
        <div
          class="rcx-box rcx-box--full rcx-modal__content  rcx-css-1vw7itl"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__content-wrapper rcx-css-56vo9k"
          >
            Modal Body
          </div>
        </div>
        <div
          class="rcx-box rcx-box--full rcx-modal__footer rcx-css-17mu816"
        >
          <div
            class="rcx-box rcx-box--full rcx-button-group--align-end rcx-button-group rcx-css-94c313"
            role="group"
          >
            <button
              class="rcx-box rcx-box--full rcx-button  rcx-button-group__item"
              type="button"
            >
              <span
                class="rcx-button--content"
              >
                Cancel
              </span>
            </button>
            <button
              class="rcx-box rcx-box--full rcx-button--primary rcx-button  rcx-button-group__item"
              type="button"
            >
              <span
                class="rcx-button--content"
              >
                Submit
              </span>
            </button>
          </div>
        </div>
      </div>
    </dialog>
  </div>
</body>
`;
